atmo/jobs/forms.py

```diff
@@ -26,9 +26,20 @@ class BaseSparkJobForm(FormControlFormMixin, CachedFileModelFormMixin,
             'class': 'identifier-taken-check',
             'data-identifier-taken-check-url': reverse_lazy('jobs-identifier-taken'),
         }),
-        help_text='A brief description of the scheduled Spark job\'s purpose, '
+        help_text='A slug-like string to identify your Spark job, '
```

Comment:
Yeah, the [job form will validate](https://github.com/mozilla/telemetry-analysis-service/blob/7ce65254adf96217931ca3f90d0b6ef9970b9e73/atmo/jobs/forms.py#L23) that with `[\w-]{1,100}$` and [remove whitespaces and underscores on the client side](https://github.com/mozilla/telemetry-analysis-service/blob/7ce65254adf96217931ca3f90d0b6ef9970b9e73/atmo/static/js/forms.js#L36-L37) when it checks for identifier availability.

So I'd say it *is* slug-like, but think the term is not easy to understand -- hence the more general term "identifier". Maybe we should call it "label-like"?